starter-code/basic-algorithms.js

```diff
@@ -1,7 +1,63 @@
 // Names and Input
+var hacker1 = "Kat";
+console.log("The driver's name is " + hacker1);
 
+var hacker2 = prompt("What is your name?");
+console.log("The navigator's name is " + hacker2);
 
 //Conditionals
 
+if (hacker1.length > hacker2.length){
+  
+  console.log("The driver has the longest name, it has " + hacker1.length);
+}
+   else if (hacker2.length > hacker1.length) {
+    
+    console.log("The navigator has the longest name, it has " + hacker2.length);
+  }
+  
+  else {
+    
+    console.log(" you both have equally long names" + hacker1.length );
+  }
+  
+  for (var x=0;x <= hacker1.length; x++) {
+  console.log(hacker1.charAt(x).toUpperCase(x));
+  }
+  
+
+  
+  function reverseString(hacker1) {
+    var splitString = hacker1.split("");
+    var reverseArray = splitString.reverse();
+    var joinArray = reverseArray.join("");
+    return joinArray;
+  }
+  console.log(reverseString(hacker1));
+  
+  // var array = [hacker1, hacker2];
+  
+  // array.sort();
+  
+  // console.log(array[0]);
+  
+  if (hacker1 < hacker2 ){
+    
+    
+    console.log("The driver's name goes first");
+    
+  }
+  
+  else if (hacker1 > hacker2){
```

Comment:
The logic is there but the syntax is not. We could use something like this:
```
var n = hacker1.localeCompare(hacker2);
if (n === 1) {
  console.log("The driver's name goes first");
} else if (n === -1) {
  console.log("Yo, the navigator goes first definitely");
} else {
  console.log("What?! You both got the same name?");
}
console.log(n);
```
Try googling .localeCompare() to see how this method works.